tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -1038,6 +1089,28 @@ class SPIRVEmitter : public ASTConsumer {
     return 0;
   }
 
+  /// Processes the given expr, casts the result into the given bool (vector)
+  /// type and returns the <result-id> of the casted value.
+  uint32_t castToBool(const Expr *expr, QualType toBoolType) {
+    // Converting to bool means comparing with value zero.
+
+    const QualType exprType = expr->getType();
+
+    // If the given expr is already a boolean or a vector of booleans, no
+    // casting is necessary.
+    if (exprType->isBooleanType() ||
+        (hlsl::IsHLSLVecType(exprType) &&
+         hlsl::GetHLSLVecElementType(exprType)->isBooleanType())) {
+      return doExpr(expr);
+    }
+
+    const spv::Op spvOp = translateOp(BO_NE, exprType);
+    const uint32_t boolType = typeTranslator.translateType(toBoolType);
+    const uint32_t originalVal = doExpr(expr);
```

Comment:
Minor: Can move this `originalVal` up since it is also used in the early return.